src/mocap_pose.cpp

```diff
@@ -242,14 +251,22 @@ rclcpp::Time MocapPose::QualisysToRosTimestamp(unsigned long long ts) {
 
 void MocapPose::WorkerThread()
 {
+    // randomize port between 6734 and maximum port.
+    // this is because if we're behind (and we should assume so) a NAT gateway, the port must not
+    // be ambiguous because there might be multiple mocap-pose clients behind the same gateway.
+    //
+    // using the same port across re-tries so QTM server in theory can consolidate or cleanup resources.
+    unsigned short udpPort = 6734 + (rand() % (65535 - 6734));
```

Comment:
It is possible that it conflicts. Good catch.

Reply:
Maybe we should have a blacklist of UDP ports?

Reply:
I didn't even think of that, good question!

The change of that is about 1/58801 and if that happens, what is the impact? Depends on the order of execution. If mavlink-router is already started then mocap_pose port allocation should fail (since it's in use already) and then I guess we'll re-try forever because we won't randomize the port on each attempt.

Are we ok with the chance of it not working? We've to also remember that we have issue logged where we'll stop using `--network=host` which means all containers get their own network namespaces in which port conflicts are not a problem.